lib/dry/types/hash/schema.rb

```diff
@@ -149,8 +149,6 @@ def resolve(hash)
           end
         end
       end
-
-      private_constant(*constants(false))
```

Comment:
This is an absolutely valid point, but I'm asking because I suspect this method _is not_ so common as `Module#private`, and I just don't know how to use it properly. I checked dependencies in one of my projects and found out that constant hiding most often used in `concurrent-ruby` gem, but those are Java guys, so no surprises. Just kidding :) Anyway, there is no absolute privacy in ruby, you always can call a method, no matter if it is protected or private, but we _used to_ mark them as private in certain cases. I do not know anything about such cases for constants, so I probably won't use them to not introduce unexpected behaviors.
I'm going to have some fun with rubinius, then I'll be able to say if we can or cannot mark that `Schema` constant as private.